galsim/table.py

```diff
@@ -280,3 +280,204 @@ def _LookupTable_hash(self):
 _galsim._LookupTable.__eq__ = _LookupTable_eq
 _galsim._LookupTable.__ne__ = lambda self, other: not self.__eq__(other)
 _galsim._LookupTable.__hash__ = _LookupTable_hash
+
+
+class LookupTable2D(object):
+    """
+    LookupTable2D represents a 2-dimensional lookup table to store function values that may be slow
+    to calculate, for which interpolating from a lookup table is sufficiently accurate.  A
+    LookupTable2D is also useful for evaluating periodic 2-d functions given samples from a single
+    period.
+
+    A LookupTable2D representing the function f(x, y) may be constructed from a list or array of `x`
+    values, a list or array of `y` values, and a 2D array of function evaluations at all
+    combinations of x and y values.  For instance:
+
+        >>> x = y = np.arange(5)
+        >>> z = x + y[:, np.newaxis]  # function is x + y, dimensions of z is (5, 5)
+        >>> tab2d = galsim.LookupTable2D(x, y, z)
+
+    To evaluate new function values with the lookup table, use the () operator:
+
+        >>> print tab2d(2.2, 3.3)
+        5.5
+
+    The () operator can also accept lists for the x and y arguments at which to evaluate the
+    LookupTable2D.  In this case, the table is evaluated for all combinations of x and y values and
+    returned as a 2D array with dimensions (len(y), len(x)).
+
+        >>> print tab2d([0, 1], [2, 3, 4])
+        [[ 2.  3.],
+         [ 3.  4.],
+         [ 4.  5.]]
+
+    Finally, if you want to just evaluate the LookupTable2D at a list of x and y points but without
+    evaluating all possible combinations of x and y values, then you can use the `scatter` keyword
+    with the () operator.
+
+        >>> print tab2d([1, 2], [3, 4], scatter=True)
```

Comment:
> So outer is ~50% faster than scatter here. 

Is there any application for which this is likely to be the tall pole for optimization?  If not, I think I'd rather just have a single meaning for what the x,y variables mean, rather than two.  (One obvious way and all that.)  But if there is anything that might really benefit from this speed up, we should keep it, at least as an option.  

I'll note that this is only called by one thing currently, which is your phase_psf class, and that uses the `scatter=True` option, so maybe we don't even need the other interface at all.  My guess is that the other places where we will want to switch to using Table2D from InterpolatedImage2D will also use the `scatter=True` version.